client/client_test.go

```diff
@@ -70,3 +78,43 @@ func TestParseHost(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdateClientVersion(t *testing.T) {
+	client := &Client{
+		transport: newMockClient(nil, func(req *http.Request) (*http.Response, error) {
+			reqString := req.URL.String()
+			splitQuery := strings.Split(reqString, "/")
+			queryVersion := splitQuery[3]
```

Comment:
I think you'd be better of using req.URL.Path instead of req.URL.String() then you don't need to worry about any leading /'s that are before the `/vX.Y` part of the URL.  And then you know that the version part is always the first thing in that string.